client/state/selectors/get-raw-site.ts

```diff
@@ -5,6 +5,10 @@ import type { AppState } from 'calypso/types';
 /**
  * Returns a raw site object by its ID.
  */
-export default function getRawSite( state: AppState, siteId: number ): SitesItem | null {
+export default function getRawSite( state: AppState, siteId: number | null ): SitesItem | null {
+	if ( ! siteId ) {
+		return null;
+	}
+
 	return getSitesItems( state )[ siteId ] || null;
```

Comment:
depending on the needs and scope of the transformation this might also be a good place to evaluate if it would be a good move to switch from `||` to `??`. It isn't always a good move because the semantics are different, but if our goal here is to detect `null` or `undefined` and then coerce to `null` then `??` might be a more appropriate idiom.